src/test/scala/com/github/music/of/the/ainur/quenya/Test.scala

```diff
@@ -49,7 +50,19 @@ class Test extends FunSuite with BeforeAndAfter {
   test("data should be exactly the same") {
     assert(diff == 0)
   }
+  val xmlData = Seq()
+  val xmlDf = spark.read
+    .option("rowTag", "foo")
+    .xml("src/test/resources/xmldata.xml")
```

Comment:
Spark infers the schema after read the whole XML, so this test is actually not test anything. You should send the two XMLs in two different interactions to be able to test this feature.

The point here is that:
```
array@array
    array$item:Stringtype
```
 Must work for the following cases:

```
<foo><bar>1</bar></foo>
<foo><bar>1</bar><bar>2</bar></foo>
```
But you need to send in two different iterations, otherwise Spark will infer an array for bar field anyway.

First send the first item of the XML with the DSL:
```
<foo><bar>1</bar></foo>
```

The DSL will try to perform an "explode_outer" and will fail because it isn't an array but it will select the element anyway using "withColumn". (This is what we want to test)

than send the second case:
```
<foo><bar>1</bar><bar>2</bar></foo>
```
The DSL will successfully do "explode_outer" on "bar" field because it's actually an array.

The whole point is that, in the load process if a single record has the case ```<foo><bar>1</bar><bar>2</bar></foo>``` it will infer array for "bar" field, but if in the whole batch just exists ```<foo><bar>1</bar></foo>``` it will infer a StringType and the DSL would fail.